src/SingularityLathe.Forge/StellarForge/Services/PlanetBuilderService.cs

```diff
@@ -30,7 +30,14 @@ public PlanetBuilderService GeneratePhysicalProperties()
 
         public PlanetBuilderService GenerateLife()
         {
-            planet.Biosphere = GetRandomBiosphere();
+            double tempLifeWeight = planet.Tempature.LifeWeight;
+            double atmosphereLifeWeight = planet.Atmosphere.LifeWeight;
+
+            double lifeWeights = tempLifeWeight + atmosphereLifeWeight;
+
+
+
+            planet.Biosphere = BioSphere.GetWeightedBiosphere();
```

Comment:
I think you need to pass in life weight to this line.  But other than that.  I think it looks good.

Reply:
Thanks for pointing that out! I've added a new commit resolving this.